### ISSUE TITLE
Envoy error: backtrace.h:104] Caught Segmentation fault, suspect faulting address 0x0

### ISSUE THREAD

**Issue by cmoulliard** [projectcontour/contour#4565]
**What steps did you take and what happened:**

The envoy proxy used by Contour is crashing and reports such errors
```
[2022-06-09 07:43:33.286][1][debug][main] [source/server/overload_manager_impl.cc:367] No overload action is configured for envoy.overload_actions.reject_incoming_connection
s.
[2022-06-09 07:43:33.287][1][info][admin] [source/server/admin/admin.cc:132] admin address: /admin/admin.sock
[2022-06-09 07:43:33.287][1][info][main] [source/server/server.cc:707] runtime: {}
[2022-06-09 07:43:33.289][1][info][config] [source/server/configuration_impl.cc:127] loading tracing configuration
[2022-06-09 07:43:33.289][1][info][config] [source/server/configuration_impl.cc:87] loading 0 static secret(s)
[2022-06-09 07:43:33.289][1][info][config] [source/server/configuration_impl.cc:93] loading 2 cluster(s)
[2022-06-09 07:43:33.289][10][debug][grpc] [source/common/grpc/google_async_client_impl.cc:50] completionThread running
[2022-06-09 07:43:33.290][1][critical][assert] [source/common/event/file_event_impl.cc:22] assert failure: SOCKET_VALID(fd).
[2022-06-09 07:43:33.290][1][critical][backtrace] [./source/server/backtrace.h:104] Caught Aborted, suspect faulting address 0x1
[2022-06-09 07:43:33.290][1][critical][backtrace] [./source/server/backtrace.h:91] Backtrace (use tools/stack_decode.py to get line numbers):
[2022-06-09 07:43:33.290][1][critical][backtrace] [./source/server/backtrace.h:92] Envoy version: a2a1e3eed4214a38608ec223859fcfa8fb679b14/1.19.1/Clean/RELEASE/BoringSSL
[2022-06-09 07:43:33.291][1][critical][backtrace] [./source/server/backtrace.h:96] #0: __restore_rt [0x7fa22946a140]
...
[2022-06-09 07:43:33.292][1][critical][backtrace] [./source/server/backtrace.h:98] #32: [0x55b3fe4e318e]
[2022-06-09 07:43:33.292][1][critical][backtrace] [./source/server/backtrace.h:98] #33: [0x55b3fe4e3303]
[2022-06-09 07:43:33.292][1][critical][backtrace] [./source/server/backtrace.h:98] #34: [0x55b3fe4df91c]
[2022-06-09 07:43:33.292][1][critical][backtrace] [./source/server/backtrace.h:96] #35: __libc_start_main [0x7fa2292b7d0a]
[2022-06-09 07:43:33.292][1][critical][backtrace] [./source/server/backtrace.h:104] Caught Segmentation fault, suspect faulting address 0x0
[2022-06-09 07:43:33.292][1][critical][backtrace] [./source/server/backtrace.h:91] Backtrace (use tools/stack_decode.py to get line numbers):
[2022-06-09 07:43:33.292][1][critical][backtrace] [./source/server/backtrace.h:92] Envoy version: a2a1e3eed4214a38608ec223859fcfa8fb679b14/1.19.1/Clean/RELEASE/BoringSSL
[2022-06-09 07:43:33.292][1][critical][backtrace] [./source/server/backtrace.h:96] #0: __restore_rt [0x7fa22946a140]
```
See gist: https://gist.github.com/cmoulliard/fc08d3194cec3375b29ce4ea3a19af7b
**Environment:**

- Contour version: 1.18.2
- Kubernetes version: (use `kubectl version`): 1.23.6
- Kubernetes installer & version: kubeadm and kubelet
- OS (e.g. from `/etc/os-release`): Fedora35

**Comment by sunjayBhatia** [comment]
Have you attempted to decode the backtrace? There have been various fixes in the 1.19.x Envoy release line that may be at play here, would have to see the actual trace to be sure

**Comment by sunjayBhatia** [comment]
Also just to be sure, since this is Contour 1.18.2, did you take note of the release notes around the previous version: https://github.com/projectcontour/contour/releases/tag/v1.18.1

There were changes to how the example manifests set up Envoy pod volumes

**Comment by cmoulliard** [comment]
> Have you attempted to decode the backtrace? There have been various fixes in the 1.19.x Envoy release line that may be at play here, would have to see the actual trace to be sure

No as I'm not a python/envoy expert ;-)

**Comment by sunjayBhatia** [comment]
this script shows usage instructions: https://github.com/envoyproxy/envoy/blob/main/tools/stack_decode.py
you also need the version of the Envoy binary you are using with debug symbols included, which you can find from the Envoy release image on dockerhub `envoyproxy/envoy-debug:<TAG>`

the easiest way i find to do this sort of thing is to do this in a docker container w/ the script above, the debug image, and your log file copied in or volume mounted

e.g. `docker run -it --rm -v <LOCAL_SRC>:<DEST> envoyproxy/envoy-build bash`

`envoyproxy/envoy-build` happens to be an image with python and addr2line etc. that is needed for decoding the stack trace, you could also use the `envoyproxy/envoy-debug` image instead and install the required tools
